test/Vault2Test.js

```diff
@@ -1,13 +1,72 @@
 const { expect } = require("chai");
-const { ethers } = require("hardhat");
+const { ethers, waffle } = require("hardhat");
+
+const provider = waffle.provider;
+
 
 describe("Vault 2", () => {
+
+	let owner, recipient, other, erc20;
+
 	beforeEach(async () => {
 		Vault2 = await ethers.getContractFactory("Vault2");
-		[owner] = await ethers.getSigners();
+		[owner, recipient, other] = await ethers.getSigners();
 
 		vault2 = await Vault2.deploy();
 
 		await vault2.deployed();
+
+		var ownerETH = await provider.getBalance(owner.address);
+		console.log('original owner eth: ', ownerETH);
+
+		
+	});
+
+	describe("mint", function () {
+
+		
+		it("Should mint 500 Vault tokens when sending 500 Eth", async function () {
+			let amount = 500;
+			// console.log('coin supply: ', totalSupply);
+			
+			// await vault2.approve(vault2.address,amount);
+			await vault2.connect(owner).mint(amount, {value: amount});
```

Comment:
You can check here if the balance of contract has increased by `amount` since you have sent ether equal to `amount`